### ISSUE TITLE
Add a UI system

### ISSUE THREAD

**Issue by jamesaorson** [ExoKomodo/Komodo#16]
A UI system should contain of, at a minimum, these elements
- Button
- Input box capable of limiting characters for input
- List of UI elements
- Panel containing UI elements
Should be able to define keyboard navigation paths from one UI element to another.
